server/odc-service/src/main/java/com/oceanbase/odc/service/iam/auth/AttemptableUsernamePasswordAuthenticationToken.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2023 OceanBase.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.oceanbase.odc.service.iam.auth;
+
+import java.util.Collection;
+
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.GrantedAuthority;
+
+import lombok.Getter;
+
+public class AttemptableUsernamePasswordAuthenticationToken extends UsernamePasswordAuthenticationToken {
```

Comment:
Your solution is kind of complex for me, could you please kindly add some detail and explanation in your PR description? For example, why do we need another authentication token?

Reply:
AttemptableUsernamePasswordAuthenticationToken is use to distinguish whether need to reset the login limit after login successful.Let me add a few more details ~